timeboost-builder/src/submit.rs

```diff
@@ -181,10 +181,18 @@ impl Sender {
                         continue;
                     }
                     let n = minicbor::len(&b);
-                    if size + n < self.size_limit {
+                    // If a single block exceeds the transaction size limit, skip it with a warning
+                    // to avoid permanently blocking the queue with an unshippable item.
+                    if n > self.size_limit {
+                        warn!(node = %self.label, block = %b.cert().data().num(), size = %n, limit = %self.size_limit, "block exceeds max transaction size; dropping");
+                        continue;
+                    }
```

Comment:
```suggestion
                    if n > self.size_limit {
                        error!(
                            node  = %self.label,
                            block = %b.cert().data().num(),
                            size  = %n,
                            limit = %self.size_limit,
                            "block exceeds max transaction size; dropping"
                        );
                        continue;
                    }
```
It is not currently reflected in the code but a single block will never exceed that max. transaction size. That being said, logging an error here is fine.